### DISCUSSION TITLE
Is it possible to use it for sending emails?

### DISCUSSION THREAD

**Discussion by gfffrtt** [a-h/templ#677] [Q&A] [answered]
Is there a way to use templ for sending emails? I'm doing a personal project and want to use it to send emails (recommendations of libs to send email are welcome too).

**Comment by joerdav** [comment, marked as answer]
Templ can generate emails, yes! It cannot send them though, you would need to use something like amazon SES or similar for the sending of the emails once generated.

**Reply by gfffrtt** [reply]
Than you so much man appreciate it, I know it might be asking for much but do you have an example, like basic, like just sending a Hello World email.

**Reply by joerdav** [reply]
No examples to hand no sorry.

Here is the docs on how to send an email with SES: https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/ses-example-send-email.html

It requires your HTML as a string to send it using SES, you can render a templ component to a string like this:

```
var b bytes.Buffer
err := myComponent(data).Render(context.Background(), &b)
if err != nil {
    // handle err
}
htmlString := b.String()
```

**Reply by gfffrtt** [reply]
Thank you so much!
